src/app/navbar/navbar.component.ts

```diff
@@ -21,27 +24,64 @@ export class NavbarComponent implements OnInit, OnDestroy {
     message: ''
   };
 
+  // Define a variable to use for showing/hiding the Login button
+  isUserLoggedIn: boolean;
+  userName: string
+  isSkip: boolean = false;
+
   @ViewChild(ShutdownModalComponent) child: ShutdownModalComponent;
 
-  constructor(private servicesHealthService: ServicesHealthService, private status: ConnectedServiceStatus, private alertService: AlertService, public ngProgress: NgProgress) { }
+  constructor(private servicesHealthService: ServicesHealthService,
+    private status: ConnectedServiceStatus,
+    private alertService: AlertService,
+    public ngProgress: NgProgress,
+    private sharedService: SharedService,
+    private authService: AuthService,
+    private cdr: ChangeDetectorRef,
+    private router: Router) {
+    // Subscribe to automatically update 
+    // "isUserLoggedIn" whenever a change to the subject is made.
+    this.sharedService.IsUserLoggedIn.subscribe(value => {
+      this.isUserLoggedIn = value.loggedIn;
+      this.userName = value.userName;
+    });
+    this.sharedService.IsLoginSkiped.subscribe(value => {
+      this.isSkip = value;
+    });
+  }
+
+  ngOnInit() { }
 
-  ngOnInit() {
+  ngAfterViewInit() {
+    // get user token from session
+    const token = sessionStorage.getItem('token');
+    const skip = sessionStorage.getItem('skip');
+    if (token != null && token.length > 0) {
+      this.isUserLoggedIn = true;
+    } else if (skip != null && skip.trim().length > 0) {
+      this.isSkip = true;
+    }
+    if(sessionStorage.getItem('userName') != null){
+      this.userName = sessionStorage.getItem('userName');
+    }
     this.start();
+    this.cdr.detectChanges();
```

Comment:
what is cdr?

Reply:
`ChangeDetectorRef` angular thing to detect changes for UI

Reply:
please add a comment